File: lib/test/structuredfield/parseParameters.test.ts
import assert from 'node:assert';
import test from 'node:test';
import { parseParameters } from '../../src/structuredfield/parse/parseParameters.js';

test('parseParameters', () => {
	assert.deepStrictEqual(parseParameters(`;a=0`), { value: { 'a': 0 }, src: `` });
	assert.deepStrictEqual(parseParameters(`;a`), { value: { 'a': true }, src: `` });
	assert.deepStrictEqual(parseParameters(`;  a;  b=?0`), { value: { 'a': true, 'b': false }, src: `` });
	assert.deepStrictEqual(parseParameters(`;a;b=?0;c=10`), { value: { 'a': true, 'b': false, 'c': 10 }, src: `` });
});

File: python/src/test/structuredfield/parseParameters_test.py
import unittest

from ...structuredfield.parse.ParsedValue import ParsedValue
from ...structuredfield.parse.parseParameters import parseParameters
from ...structuredfield.SfException import SfException


class TestParseToken(unittest.TestCase):
    def test_parseString(self):
        self.assertEqual(parseParameters(";a=0"), ParsedValue({"a": 0}, ""))
        self.assertEqual(parseParameters(";a"), ParsedValue({"a": True}, ""))
        self.assertEqual(
            parseParameters(";  a;  b=?0"), ParsedValue({"a": True, "b": False}, "")
        )
        self.assertEqual(
            parseParameters(";a;b=?0;c=10"),
            ParsedValue({"a": True, "b": False, "c": 10}, ""),
        )
